lua/dap.lua

```diff
@@ -845,24 +860,68 @@ function M.restart()
   end
 end
 
-function M.toggle_breakpoint(condition)
+function M.list_breakpoints(open_quickfix)
+  local bp_signs = get_breakpoint_signs()
+  local num_bufs = #bp_signs
+  local qf_list = {}
+  for _, buf_bp_signs in pairs(bp_signs) do
+    local bufnr = buf_bp_signs.bufnr
+    for _, bp in pairs(buf_bp_signs.signs) do
+      local condition = bp_info[bp.id].condition;
+      local hitCondition = bp_info[bp.id].hitCondition;
+      local logMessage = bp_info[bp.id].logMessage;
+      local text = table.concat(vim.tbl_filter(function(t) return t end, {
+               unpack(api.nvim_buf_get_lines(bufnr, bp.lnum - 1, bp.lnum, false), 1),
```

Comment:
I may be missing something,  but it looks like the tbl_filter call here is a noop and could be removed?

Reply:
It filters out `nil` from the table, otherwise I remember getting an error that `nil` can not be concatenated.

Reply:
You were right. Maybe the error I remember was caused by something different. If we ever observe an error here we can still add the filtering again.

In this case, `nil` shouldn't cause any problems. But it still did: https://github.com/nvim-treesitter/nvim-treesitter/pull/341#event-3689877983